src/families/cosmos/bridge/libcore.js

```diff
@@ -0,0 +1,237 @@
+      ) {
+        return new CosmosRedelegationInProgress();
+      }
+    }
+  }
+  return null;
+};
+
+const getMaxEstimatedBalance = (a, t, estimatedFees) => {
+  const { cosmosResources } = a;
+  let blockBalance = BigNumber(0);
+  if (cosmosResources) {
+    blockBalance = cosmosResources.pendingRewardsBalance
+      .plus(cosmosResources.unboundingBalance)
+      .plus(cosmosResources.delegatedBalance);
+  }
+
+  return a.balance.minus(estimatedFees).minus(blockBalance);
+};
+
+const getTransactionStatus = async (a, t) => {
+  const errors = {};
+  const warnings = {};
+
+  if (t.mode === "redelegate") {
+    const redelegationError = redelegationStatusError(a, t);
+    if (redelegationError) {
+      errors.redelegation = redelegationError;
+    }
+  } else if (t.mode === "undelegate") {
+    if (a.cosmosResources && a.cosmosResources.unbondings.length >= 7) {
+      errors.unbondings = new CosmosTooMuchUnboundings();
+    }
+  } else if (
+    ["delegate", "claimReward", "claimRewardCompound"].includes(t.mode)
+  ) {
+    if (
+      t.validators.some(
+        (v) => !v.address || !v.address.includes("cosmosvaloper")
+      )
+    )
+      errors.recipient = new InvalidAddress(null, {
+        currencyName: a.currency.name,
+      });
+  } else {
+    if (a.freshAddress === t.recipient) {
+      errors.recipient = new InvalidAddressBecauseDestinationIsAlsoSource();
+    } else {
+      const { recipientError, recipientWarning } = await validateRecipient(
+        a.currency,
+        t.recipient
+      );
+
+      if (recipientError) {
+        errors.recipient = recipientError;
+      }
+
+      if (recipientWarning) {
+        warnings.recipient = recipientWarning;
+      }
+    }
+  }
+
+  let estimatedFees = BigNumber(0);
+  let amount = t.amount;
+
+  if (!errors.recipient) {
+    await calculateFees(a, t).then(
+      (res) => {
+        estimatedFees = res.estimatedFees;
+      },
+      (error) => {
+        if (error.name === "NotEnoughBalance") {
+          errors.amount = error;
+        } else {
+          throw error;
+        }
+      }
+    );
+  }
+
+  if (!amount && t.mode !== "send") {
+    amount = t.validators.reduce(
+      (old, current) => old.plus(current.amount),
+      BigNumber(0)
+    );
+  }
+
+  let totalSpent = !t.useAllAmount
+    ? amount.plus(estimatedFees)
+    : getMaxEstimatedBalance(a, t, estimatedFees);
+
+  if (
+    !errors.recipient &&
+    !errors.amount &&
+    (amount.lt(0) || totalSpent.gt(a.balance))
+  ) {
+    errors.amount = new NotEnoughBalance();
+    totalSpent = BigNumber(0);
+    amount = BigNumber(0);
+  }
+
+  return Promise.resolve({
+    errors,
+    warnings,
+    estimatedFees,
+    amount,
+    totalSpent,
+  });
+};
+
+const prepareTransaction = async (a, t) => {
+  return t;
+};
+
+const currencyBridge: CurrencyBridge = {
+  preload: async () => {
+    const validators = await getValidators();
+    setCosmosPreloadData({ validators });
+    return { validators };
```

Comment:
please make it explicit that preload function returns a Promise of CosmosPreloadData (as in mock)